plugins/cr8-copy-s3/src/main/java/io/crate/copy/s3/S3FileInputFactory.java

```diff
@@ -23,15 +23,31 @@
 
 import io.crate.execution.engine.collect.files.FileInput;
 import io.crate.execution.engine.collect.files.FileInputFactory;
+import io.crate.types.DataTypes;
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.common.settings.Settings;
 
 import java.net.URI;
+import java.util.function.Function;
 
 public class S3FileInputFactory implements FileInputFactory {
 
     public static final String NAME = "s3";
+    static final Setting<String> PROTOCOL = new Setting<>(
```

Comment:
Could also turn this into a `S3Protocol` enum instead of a string. To signal that there are only 2 possible values and to model at which point it got parsed.  But fine this way as well.